app/code/Magento/InventoryShipping/Model/PriorityShippingAlgorithm.php

```diff
@@ -0,0 +1,279 @@
+        foreach ($order->getItems() as $orderItem) {
+            $itemSku = $orderItem->getSku();
+            $qtyToDeliver = $orderItem->getQtyOrdered();
+
+            //check if order item is not delivered yet
+            if ($orderItem->isDeleted() || $orderItem->getParentItemId() || $this->isZero((float)$qtyToDeliver)) {
+                continue;
+            }
+
+            foreach ($sources as $source) {
+                if (!$source->isEnabled()) {
+                    continue;
+                }
+
+                $sourceItem = $this->getStockItemBySku($source->getSourceCode(), $itemSku);
+                $sourceItemQty = $sourceItem->getQuantity();
+                $qtyToDeduct = min($sourceItemQty, $qtyToDeliver);
+
+                // check if source has some qty of SKU, so it's possible to take them into account
+                if ($this->isZero((float)$sourceItemQty)) {
+                    continue;
+                }
+
+                $sourceItemSelection = $this->sourceItemSelectionFactory->create(
+                    [
+                        'sku' => $itemSku,
+                        'qty' => $qtyToDeduct,
+                        'qtyAvailable' => $sourceItemQty,
+                    ]
+                );
+
+                $sourceItemSelections = $this->updateSourceItemSelections(
+                    $sourceItemSelections,
+                    $sourceItemSelection,
+                    $sourceItem
+                );
+
+                $qtyToDeliver -= $qtyToDeduct;
+            }
+
+            // if we go throw all sources from the stock and there is still some qty to delivery,
+            // then it doesn't have enough items to delivery
+            if (!$this->isZero($qtyToDeliver)) {
+                $isShippable = false;
+            }
+        }
+
+        $sourceSelections = $this->createSourceSelection($sourceItemSelections);
+
+        return $this->shippingAlgorithmResultFactory->create([
+            'sourceSelections' => $sourceSelections,
+            'isShippable' => $isShippable
+        ]);
+    }
+
+    /**
+     * Retrieve sources are related to current stock that are ordered by priority
+     *
+     * @param int $storeId
+     *
+     * @return SourceInterface[]
+     */
+    private function getSourcesByStoreId(int $storeId): array
+    {
+        $store = $this->storeManager->getStore($storeId);
+        $website = $this->websiteRepository->getById($store->getWebsiteId());
+        $stock = $this->stockResolver->get(SalesChannelInterface::TYPE_WEBSITE, $website->getCode());
+
+        return $this->getAssignedSourcesForStock->execute((int)$stock->getStockId());
+    }
+
+    /**
+     * Retrieve stock item from specific source by SKU
+     *
+     * @param string $stockCode
+     * @param string $sku
+     *
+     * @return SourceItemInterface
+     */
+    private function getStockItemBySku(string $stockCode, string $sku): SourceItemInterface
```

Comment:
Description talks about one behavior
Method name talks about another behavior
Implementation return the other

Also, you can reuse `\Magento\Inventory\Model\SourceItem\Command\GetSourceItemsBySkuInterface`